android/src/play/java/com/dooboolab/RNIap/RNIapModule.java

```diff
@@ -232,14 +224,8 @@ public void flushFailedPurchasesCachedAsPending(final Promise promise) {
         new Runnable() {
           @Override
           public void run() {
-            final WritableNativeArray array = new WritableNativeArray();
             Purchase.PurchasesResult result =
                 billingClient.queryPurchases(BillingClient.SkuType.INAPP);
-            if (result == null) {
-              // No results for query
-              promise.resolve(false);
-              return;
-            }
```

Comment:
Same here. How about putting `@NonNull` decorator?

Reply:
Same as in the other one, `array` is created but never used and `result==null` is never true as `queryPurchase` return is non-nullable

Reply:
Thanks for the reply!! Understood 👍